modules/process/manager_stacktraces.go

```diff
@@ -0,0 +1,256 @@
+				continue
+			}
+			if len(value) != 1 {
+				// Unexpected...
+				return nil, 0, fmt.Errorf("label: %s in goroutine stack with unexpected number of values: %v", name, value)
+			}
+
+			stack.Labels = append(stack.Labels, &Label{Name: name, Value: value[0]})
+		}
+
+		stack.Count = sample.Value[0]
+		goroutineCount += stack.Count
+
+		// Now get the processStack for this goroutine sample
+		var processStack *Process
+		if pidvalue, ok := sample.Label[PIDPProfLabel]; ok && len(pidvalue) == 1 {
+			pid := IDType(pidvalue[0])
+			processStack, ok = pidMap[pid]
+			if !ok && pid != "" {
+				ppid := IDType("")
+				if value, ok := sample.Label[PPIDPProfLabel]; ok && len(value) == 1 {
+					ppid = IDType(value[0])
+				}
+				description := "(dead process)"
+				if value, ok := sample.Label[DescriptionPProfLabel]; ok && len(value) == 1 {
+					description = value[0] + " " + description
+				}
+				ptype := "code"
+				if value, ok := sample.Label[ProcessTypePProfLabel]; ok && len(value) == 1 {
+					stack.Labels = append(stack.Labels, &Label{Name: ProcessTypePProfLabel, Value: value[0]})
+				}
+				processStack = &Process{
+					PID:         pid,
+					ParentPID:   ppid,
+					Description: description,
+					Type:        ptype,
+				}
+
+				pidMap[processStack.PID] = processStack
+				added := false
+				if processStack.ParentPID != "" && !flat {
+					if parent, ok := pidMap[processStack.ParentPID]; ok {
+						parent.Children = append(parent.Children, processStack)
+						added = true
+					}
+				}
+				if !added {
+					processStacks = append(processStacks, processStack)
+				}
+			}
+		}
+		if processStack == nil {
+			var ok bool
+			processStack, ok = pidMap[""]
+			if !ok {
+				processStack = &Process{
+					Description: "(unassociated)",
+					Type:        "code",
+				}
+				pidMap[processStack.PID] = processStack
+				processStacks = append(processStacks, processStack)
+			}
+		}
+
+		// Now walk through the locations...
+		for _, location := range sample.Location {
+			for _, line := range location.Line {
+				entry := &StackEntry{
+					Function: line.Function.Name,
+					File:     line.Function.Filename,
+					Line:     int(line.Line),
+				}
+				stack.Entry = append(stack.Entry, entry)
+			}
+		}
+		stack.Description = "(unknown)"
+		if len(stack.Entry) > 0 {
+			stack.Description = stack.Entry[len(stack.Entry)-1].Function
+		}
+
+		processStack.Stacks = append(processStack.Stacks, stack)
+	}
+
+	if onlyRequests {
+		var requestStacks []*Process
+		i := len(processStacks) - 1
+		for i >= 0 {
+			processStack := processStacks[i]
+			if processStack.Type == RequestProcessType {
+				requestStacks = append(requestStacks, processStack)
+				i--
+				continue
+			}
+			if len(processStack.Children) > 0 {
+				processStacks = processStacks[:i]
+				processStacks = append(processStacks, processStack.Children...)
```

Comment:
```suggestion
				processStacks = append(processStacks[:i], processStack.Children...)
```
Seems like it can be made into 1 expression.

Reply:
I'm surprised that lint tolerates that. It has specific tastes on append... however, it appears to pass lint so I will accept the suggestion.